evap/staff/tests/test_views.py

```diff
@@ -2396,6 +2415,19 @@ def test_state_change_log_translated(self, trans):
         response = self.app.get(self.url, user=self.manager)
         self.assertContains(response, "TRANSLATED-state: TRANSLATED-new &#8594; TRANSLATED-prepared")
 
+    def test_evaluation_confirm_without_main_language(self):
+        response = self.app.get(self.url, user=self.manager, status=200)
+        form = response.forms["evaluation-form"]
+
+        form["main_language"] = "x"
+        response = form.submit("operation", value="approve")
+        self.assertContains(response, "You have to set a main language to approve this evaluation.")
+
+        self.assertNotEqual(Evaluation.objects.first().state, self.evaluation.State.APPROVED)
+
+        form["main_language"] = "en"
+        response = form.submit("operation", value="approve", status=302)
```

Comment:
The function just ends here without touch `response` again - can you add one more assert?

Reply:
There is notting to assert as it's a redirect.